web/authentication/x509/x509.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package x509
+
+import (
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/exporter-toolkit/web/authentication"
+)
+
+type RequireClientCertsFunc func() bool
+
+type VerifyOptionsFunc func() x509.VerifyOptions
+
+type VerifyPeerCertificateFunc func([][]byte, [][]*x509.Certificate) error
```

Comment:
In general, by documenting this do you mean adding comments for the functions/interfaces you specifically pointed to, or rather adding a page/section to `docs/`? Or both?

Reply:
I removed the types for these particular functions in the process but I added comments to the structure and Authenticate func to describe the intent and reference the counterparts from crypto/tls that this is implementation is mimicking. I believe it will also make it easier to keep it up to date.